lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -182,49 +184,83 @@
     data_disks:
         description:
             - Describes list of data disks.
+            - This list cannot be changed after creation currently, use M(azure_rm_manageddisk) to manage the data disk.
         version_added: "2.4"
         suboptions:
             lun:
                 description:
-                    - The logical unit number for data disk
-                default: 0
+                    - The logical unit number for data disk.
+                    - This value is used to identify data disks within the VM and therefore must be unique for each data disk attached to a VM.
+                required: true
                 version_added: "2.4"
             disk_size_gb:
                 description:
-                    - The initial disk size in GB for blank data disks
+                    - The initial disk size in GB for blank data disks.
+                    - This value cannot be larger than 1023 GB.
```

Comment:
it's so hard to understand logic,  when os disk  is manged disk, data disk is managed  disk. etc.. can we add detail samples to demo how to do it? a special section for disk, it could be  very helpful to users.